Frameworks/CoreGraphics/CGColorSpace.mm

```diff
@@ -91,70 +92,77 @@ CGColorSpaceRef CGColorSpaceCreateIndexed(id baseSpace, int lastIndex, void* col
  @Status Interoperable
 */
 CGColorSpaceRef CGColorSpaceCreateDeviceRGB() {
-    return (CGColorSpaceRef) new __CGColorSpace(_ColorRGBA);
+    return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelRGB);
 }
 
 /**
  @Status Stub
- @Notes Always returns RGBA colorspace
+ @Notes Always returns RGB colorspace
 */
 CGColorSpaceRef CGColorSpaceCreatePattern(CGColorSpaceRef source) {
     UNIMPLEMENTED();
-    return (CGColorSpaceRef) new __CGColorSpace(_ColorRGBA);
+    return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelRGB);
 }
 
 /**
- @Status Stub
- @Notes Always returns RGB colorspace
+ @Status Interoperable
 */
-CGColorSpaceModel CGColorSpaceGetModel(CGColorSpaceRef colorSpace) {
-    UNIMPLEMENTED();
-    TraceWarning(TAG, L"CGColorSpaceGetModel not implemented");
-    return kCGColorSpaceModelRGB;
+CGColorSpaceRef CGColorSpaceCreateDeviceGray() {
+    return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelMonochrome);
 }
 
 /**
  @Status Caveat
- @Notes Doesn't support all bitmap formats
+ @Notes Only GenericRGB, GenericRGBLinear and GenericGray supported
 */
-size_t CGColorSpaceGetNumberOfComponents(CGColorSpaceRef pSpace) {
-    switch (((__CGColorSpace*)pSpace)->colorSpace) {
-        case _ColorRGBA:
-        case _ColorARGB:
-            return 3;
+CGColorSpaceRef CGColorSpaceCreateWithName(CFStringRef name) {
+    const char* strName = static_cast<const char*>([(__bridge NSString*)name UTF8String]);
 
-        case _ColorGrayscale:
-            return 1;
-
-        default:
-            assert(0);
-            return 0;
+    if (strcmp(strName, "kCGColorSpaceGenericRGB") == 0) {
+        return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelRGB);
```

Comment:
It probably makes sense to keep a dictionary (CFDictionary) of __CGColorSpace instances and return the instance instead of creating new ones everytime.  Clearly, this is just an optimization.